File: server/src/main/java/gg/mineral/server/network/packet/play/clientbound/EntityPropertiesPacket.java
package gg.mineral.server.network.packet.play.clientbound;

import java.util.Map;

import gg.mineral.api.entity.attribute.Attribute;
import gg.mineral.api.network.packet.Packet;
import io.netty.buffer.ByteBuf;

public final record EntityPropertiesPacket(int entityId, Map<String, Attribute.Property> properties)
        implements Packet.OUTGOING {
    @Override
    public void serialize(ByteBuf os) {
        os.writeInt(entityId);
        os.writeInt(properties.size());
        writeProperties(os, properties);
    }

    @Override
    public byte getId() {
        return 0x20;
    }

}

File: server/src/main/kotlin/gg/mineral/server/network/packet/play/clientbound/EntityPropertiesPacket.kt
package gg.mineral.server.network.packet.play.clientbound

import gg.mineral.api.entity.attribute.Attribute
import gg.mineral.api.network.packet.Packet
import io.netty.buffer.ByteBuf

@JvmRecord
data class EntityPropertiesPacket(val entityId: Int, val properties: Map<String, Attribute.Property>) :
    Packet.OUTGOING {
    override fun serialize(os: ByteBuf) {
        os.writeInt(entityId, properties.size)
        os.writeProperties(properties)
    }

    override val id: Byte
        get() = 0x20
}
